resin/models/device.py

```diff
@@ -878,3 +871,90 @@ def generate_device_key(self, uuid):
             '/api-key/device/{id}/device-key'.format(id=device_id), 'POST',
             endpoint=self.settings.get('api_endpoint')
         )
+
+    def get_dashboard_url(self, uuid):
+        """
+        Get Resin Dashboard URL for a specific device.
+
+        Args:
+            uuid (str): device uuid.
+
+        Examples:
+            >>> resin.models.device.get_dashboard_url('19619a6317072b65a240b451f45f855d')
+            https://dashboard.resin.io/devices/19619a6317072b65a240b451f45f855d/summary
+
+        """
+
+        if not uuid:
+            raise ValueError("Device UUID must be a non empty string")
+        dashboard_url = self.settings.get('api_endpoint').replace('api', 'dashboard')
+        return urljoin(
+            dashboard_url,
+            '/devices/{}/summary'.format(uuid)
+        )
+
+    def grant_support_access(self, uuid, expiry_timestamp):
+        """
+        Grant support access to a device until a specified time.
+
+        Args:
+            uuid (str): device uuid.
+            expiry_timestamp (int): a timestamp in ms for when the support access will expire.
+
+        Returns:
+            OK.
+
+        Examples:
+            >> > resin.models.device.grant_support_access('49b2a76b7f188c1d6f781e67c8f34adb4a7bfd2eec3f91d40b1efb75fe413d', 1511974999000)
+            'OK'
+
+        """
+
+        if not expiry_timestamp or expiry_timestamp <= int((datetime.utcnow() - datetime.utcfromtimestamp(0)).total_seconds() * 1000):
+            raise exceptions.InvalidParameter('expiry_timestamp', expiry_timestamp)
+
+        device_id = self.get(uuid)['id']
+        params = {
+            'filter': 'id',
+            'eq': device_id
+        }
+
+        data = {
+            'is_accessible_by_support_until__date': expiry_timestamp
+        }
+
+        return self.base_request.request(
+            'device', 'PATCH', params=params, data=data,
+            endpoint=self.settings.get('pine_endpoint')
+        )
+
+    def revoke_support_access(self, uuid):
+        """
+        Revoke support access to a device.
+
+        Args:
+            uuid (str): device uuid.
+
+        Returns:
+            OK.
+
+        Examples:
+            >> > resin.models.device.revoke_support_access('5685')
```

Comment:
I guess this example ought to be using a uuid rather than an app_id ?

Reply:
@lurch ah yes, my bad. Thanks for pointing this out, updated!